barretenberg/cpp/src/CMakeLists.txt

```diff
@@ -26,6 +26,8 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 18)
         # We target clang18 and need this, eventually warning should be fixed or this will be unconditional.
         add_compile_options(-Wno-vla-cxx-extension)
+        # This gets in the way of a valid designated initializer pattern (i.e. MyClass my_class{ .my_member = init_value })
```

Comment:
This was added after a discussion with Adam. IIUC this is part of Charlies incoming PR anyway.